src/devices/Ads1115/Ads1115.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Ads1115
+{
+    /// <summary>
+    /// Analog-to-Digital Converter ADS1115
+    /// </summary>
+    public class Ads1115 : IDisposable
+    {
+        private I2cDevice _sensor = null;
+
+        private readonly byte _adcMux;
+        private readonly byte _adcPga;
+        private readonly byte _adcRate;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        /// <param name="mux">Input Multiplexer</param>
+        /// <param name="pga">Programmable Gain Amplifier</param>
+        /// <param name="rate">Data Rate</param>
+        public Ads1115(I2cDevice sensor, InputMultiplexeConfig mux = InputMultiplexeConfig.AIN0, PgaConfig pga = PgaConfig.FS4096, DataRate rate = DataRate.SPS128)
+        {
+            _sensor = sensor;
+            _adcMux = (byte)mux;
+            _adcPga = (byte)pga;
+            _adcRate = (byte)rate;
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initialize ADS1115
+        /// </summary>
+        private void Initialize()
+        {
+            byte configHi = (byte)((_adcMux << 4) +
+                            (_adcPga << 1) +
+                            (byte)DeviceMode.Continuous);
+
+            byte configLo = (byte)((_adcRate << 5) +
+                            ((byte)(ComparatorMode.Traditional) << 4) +
+                            ((byte)ComparatorPolarity.Low << 3) +
+                            ((byte)ComparatorLatching.NonLatching << 2) +
+                            (byte)ComparatorQueue.Disable);
+
+            _sensor.Write(new [] { (byte)Register.ADC_CONFIG_REG_ADDR, configHi, configLo });
+        }
+
+        /// <summary>
+        /// Read Raw Data
+        /// </summary>
+        /// <returns>Raw Value</returns>
+        public short ReadRaw()
+        {
+            short val;
+            Span<byte> data = stackalloc byte[2];
+
+            _sensor.Write(new [] { (byte)Register.ADC_CONVERSION_REG_ADDR });
+            _sensor.Read(data);
+
+            data.Reverse();
+            val = BitConverter.ToInt16(data.ToArray(), 0);
+
+            return val;
+        }
+
+        /// <summary>
+        /// Convert Raw Data to Voltage
+        /// </summary>
+        /// <param name="val">Raw Data</param>
+        /// <returns>Voltage</returns>
+        public double RawToVoltage(short val)
+        {
+            double voltage;
+            double resolution;
+
+            if (_adcPga == 0x00)
```

Comment:
We should convert this to a switch statement. Also, you can either cast _adcPga to a PgaConfig so that you can do comparisons like: `case (PgaConfig.FS6144)` etc, or you can also simply say that _adcPga is of type PgaConfig instead of it being a byte.